modules/002-deckhouse/hooks/check_kernel_versions.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2022 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This hook checks nodes kernel requirements and set internal flag stopMainQueue.
+This flag used in another hook, stop_main_queue.go, which stops main queue if flag is true.
+We cannot stop queue in this hook, because we loose metrics if hook fails.
+*/
+
+package hooks
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook/metrics"
+	"github.com/flant/addon-operator/sdk"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/deckhouse/deckhouse/go_lib/set"
+)
+
+type nodeKernelVersion struct {
+	Name          string
+	KernelVersion string
+}
+
+const (
+	ciliumConstraint            = ">= 4.9.17"
+	ciliumAndIstioConstraint    = ">= 5.7"
+	nodeKernelCheckMetricsGroup = "node_kernel_check"
+	nodeKernelCheckMetricName   = "d8_node_kernel_does_not_satisfy_requirements"
+)
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnAfterHelm: &go_hook.OrderedConfig{Order: 5},
```

Comment:
Add a comment, why it is important to run the hook OnAfterHelm.

Reply:
Why it isn't enough to run on k8s event?

Reply:
it was important when we stops main queue, now it is not needed.